Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -1075,12 +1075,27 @@ public final class PackageBuilder {
             case .library, .test:
                 break
             case .executable:
-                let executableTargets = targets.filter({ $0.type == .executable })
-                if executableTargets.count != 1 {
-                    diagnostics.emit(
-                        .invalidExecutableProductDecl(product.name),
-                        location: diagnosticLocation()
-                    )
+                let executableTargetCount = targets.filter { $0.type == .executable }.count
+                if executableTargetCount != 1 {
+                    if executableTargetCount == 0 {
```

Comment:
I wonder if the code will be easier to understand if we use `switch`

Reply:
I tried a switch, but it didn't really improve it as we still need to if let on sum_only.